juniper_official/Paragon/Foghorn/l3vpn-icmp.rule

```diff
@@ -0,0 +1,123 @@
+/*
+ * Collects ICMP probe status between PE and CE for a given routing-instance;
+ */
+healthbot {
+    topic protocol.icmp {
+        description "Collect routing instance and interfaces stats and notify anomalies";
+        synopsis "L3VPN session state analyzer";
+        rule check-icmp-statistics-with-source {
+            keys host;
+            sensor icmp {
+                synopsis "iAgent sensor definition";
+                description "Ping rpc to collect data from network device";
+                iAgent {
+                    file icmp-statistics-with-source.yml;
+                    table pingWithSourceTable;
+                    frequency 60s;
+                    args count {
+                        arg-value "{{count-var}}";
+                    }
+                    args host {
+                        arg-value "{{host-var}}";
+                    }
+                    args routing-instance {
+                        arg-value "{{rt-instance}}";
+                    }
+                    args source {
+                        arg-value "{{source-var}}";
+                    }
+                }
+            }
+            field host {
+                sensor icmp {
+                    path host;
+                }
+                type string;
+                description "Destination address/Host address in ICMP probe";
+            }
+            field packet-loss-percent {
+                sensor icmp {
+                    path packet-loss;
+                }
+                type integer;
+                description "ICMP probe packet loss percentage"
+            }
+            trigger packet-loss {
+                synopsis "ICMP packet loss KPI";
+                description "Sets health based on the packet loss";
+                frequency 1offset;
+                term is-device-not-reachable {
+                    when {
+                        equal-to "$packet-loss-percent" 100 {
+                            time-range 2offset;
+                        }
+                    }
+                    then {
+                        status {
+                            color red;
+                            message "Host $host not reachable";
```

Comment:
Add loss percentage in message i.e.  "Host $host not reachable. $packet-loss-percent % packet loss"